core/drivers/stm32_bsec.c

```diff
@@ -160,6 +160,30 @@ static uint32_t bsec_status(void)
 	return io_read32(bsec_base() + BSEC_OTP_STATUS_OFF);
 }
 
+static bool state_is_invalid_mode(void)
+{
+	return bsec_status() & BSEC_MODE_INVALID;
+}
+
+static bool state_is_secured_mode(void)
+{
+	return bsec_status() & BSEC_MODE_SECURED;
+}
+
+static bool state_is_closed_mode(void)
+{
+	uint32_t close_mode = 0;
+	const uint32_t mask = CFG0_CLOSED_MASK;
+
+	if (IS_ENABLED(CFG_STM32MP13))
+		return bsec_status() & BSEC_MODE_CLOSED;
+
+	if (stm32_bsec_read_otp(&close_mode, CFG0_OTP))
+		panic("Unable to read OTP");
+
+	return (close_mode & mask) == mask;
```

Comment:
Maybe simplify here: `return close_mode & BIT(CFG0_OTP_SECURED_POS);`

Reply:
Ok, looking back at the code, CFG0_OTP_SECURED_POS is not "secured position" in the bitfield. BIT(CFG0_OTP_SECURED_POS) is set if the device is secure and closed and is not set otherwise.

Changing CFG0_OTP_SECURED_POS to CFG0_OTP_CLOSED_DEVICE and remove now useless CFG0_CLOSED_MASK